content/overview/quick-start.md

```diff
@@ -23,80 +23,126 @@ For more details, refer to our documentation.
 
 {{< button url="https://docs.haystack.deepset.ai/docs/installation" text="Docs: Installation" color="green">}}
 
-## Ask Questions to a Webpage
+## Prerequisites
 
-This is a very simple pipeline that can answer questions about the contents of a webpage. It uses `gpt-4o-mini` with the `OpenAIGenerator`.
+You need an [OpenAI API Key](https://platform.openai.com/api-keys) and a [SerperDev API Key](https://serper.dev/api-key)
 
-Run the following **Quickstart** or the equivalent **Corresponding Pipeline** below. See the pipeline visualized in **Pipeline Graph**.
-
-{{< tabs totalTabs="2">}}
+```python
+import os
 
-{{< tab tabName="Quickstart: Chat with Website Pipeline"  >}}
-First, install Haystack:
-```bash
-pip install haystack-ai trafilatura
+os.environ["OPENAI_API_KEY"] = "<YOUR OPENAI API KEY>"
+os.environ["SERPERDEV_API_KEY"] = "<YOUR SERPERDEV API KEY>"
 ```
 
-```python
-import os
+## 🤖 Basic Agent with Haystack
 
-from haystack import Pipeline
-from haystack.components.fetchers import LinkContentFetcher
-from haystack.components.converters import HTMLToDocument
-from haystack.components.builders import ChatPromptBuilder
+You can build a working agent in just a few lines with the [Agent](https://docs.haystack.deepset.ai/docs/agent) component. It takes in a user question, decides whether to use a tool (like web search), and returns a response without manual routing.
+
+Below is a minimal example using [SerperDevWebSearch](https://docs.haystack.deepset.ai/docs/serperdevwebsearch) component as a tool and OpenAI's `gpt-4o-mini` chat model with [OpenAIChatGenerator](https://docs.haystack.deepset.ai/docs/openaichatgenerator):
+
+```python
+from haystack.components.agents import Agent
 from haystack.components.generators.chat import OpenAIChatGenerator
 from haystack.dataclasses import ChatMessage
+from haystack.tools import ComponentTool
+from haystack.components.websearch import SerperDevWebSearch
 
-os.environ["OPENAI_API_KEY"] = "Your OpenAI API Key"
+search_tool = ComponentTool(component=SerperDevWebSearch())
 
-fetcher = LinkContentFetcher()
-converter = HTMLToDocument()
-prompt_template = [
-    ChatMessage.from_user(
-      """
-      According to the contents of this website:
-      {% for document in documents %}
-        {{document.content}}
-      {% endfor %}
-      Answer the given question: {{query}}
-      Answer:
-      """
-    )
-]
+basic_agent = Agent(
+    chat_generator=OpenAIChatGenerator(model="gpt-4o-mini"),
```

Comment:
We could leave a comment here to mention that any other ChatGenerator will work the same way and they can be easily swapped.